src/SilbinaryWolf/Components/ComponentService.php

```diff
@@ -185,6 +186,33 @@ public function generateTemplateCode(array $res, $parser)
         return $result;
     }
 
+    /**
+     * Recursively replace nonassociative arrays with ArrayListExportable and
+     * output with 'var_export' to produce template logic for the nested data.
+     *
+     * @param array     $array      The nested data to export
+     * @param bool      $root       Ignore this
+     * @return string               Executable template logic
+     */
+    private static function exportNestedDataForTemplates(array $array, $root = true)
```

Comment:
It *could* be a method, but as a static function it's clear that it doesn't interact with `$this` at all, it takes nestsed data, recurses it, and spits back the output. Whilst I think this is technically better/clearer, I'm happy to change it if you prefer.

Reply:
Hmm, while I don't disagree, if we're going to make it static I would lean towards making it public so it can be accessed from outside the service class without needing to instantiate a service. But for consistency purposes, it should probably stay public and non-static. Happy to leave this one to your judgement, just let me know what you decide on :)

Reply:
My vote is:
- private: because it's too specific and niche to be externally useful. Else I would be all for making it public.
- static: because it works independant of `$this` which is context I would find useful when navigating unfamiliar code, etc.

But I'm the noob here, so happy to be overridden.

Reply:
As discussed IRL, I can always see someone trying to access a private function, no matter how niche it is :) But I agree, private and static makes the most sense!